dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -604,27 +657,58 @@ private void writeTextAnnotation(BratAnnotationDocument aDoc, AnnotationFS aFS)
     {
         String superType = getBratType(aFS.getCAS().getTypeSystem().getParent(aFS.getType()));
         String type = getBratType(aFS.getType());
-        
-        BratTextAnnotation anno = new BratTextAnnotation(nextTextAnnotationId, type,
-                aFS.getBegin(), aFS.getEnd(), aFS.getCoveredText());
-        nextTextAnnotationId++;
-
-        conf.addEntityDecl(superType, type);
-        
-        conf.addLabelDecl(anno.getType(), aFS.getType().getShortName(), aFS.getType()
-                .getShortName().substring(0, 1));
-
-        if (!conf.hasDrawingDecl(anno.getType())) {
-            conf.addDrawingDecl(new BratTextAnnotationDrawingDecl(anno.getType(), "black",
-                    palette[nextPaletteIndex % palette.length]));
-            nextPaletteIndex++;
+        // check if the type has  a value to display, replace the the type by the value
+        // do it similar as with declarations that looks for the value
+        LinkedList<String> spanDatas=null;
+        String value=type;
+        if (spanTypes.containsKey(aFS.getType().getName())) {
+        	spanDatas=spanTypes.get(aFS.getType().getName());
+        } else  if (spanTypes.containsKey(aFS.getCAS().getTypeSystem().getParent(aFS.getType()).getName() ))  {
+        	spanDatas=spanTypes.get(aFS.getCAS().getTypeSystem().getParent(aFS.getType()).getName() );
+        }
+        for (String spanData : spanDatas){	
+	       try {
+	        	if (!spanData.equalsIgnoreCase("")){
+	        		String [] splits=spanData.split("\\|");
+	        		if (splits.length>1){
+	        			FeatureStructure currentAnnot = aFS.getFeatureValue(aFS.getType().getFeatureByBaseName(splits[0]));
+	        			for (int f=1;f<splits.length-1;f++){     		
+		        		currentAnnot = currentAnnot.getFeatureValue(currentAnnot.getType().getFeatureByBaseName(splits[f]));
+		        	}
+		        	value= currentAnnot.getFeatureValueAsString(currentAnnot.getType().getFeatureByBaseName(splits[splits.length-1]));
+		        }  	else {
+		        	value=aFS.getFeatureValueAsString(aFS.getType().getFeatureByBaseName(splits[0]));
+		        	}
+		        }
+	        } catch (Exception E){
```

Comment:
What kind of exception are you expecting here? Normally, if there is a problem that cannot be handled, the exception should be allows to fail the execution.

Reply:
I think that there were some cases (the ROOT of the dpendencies ) that the getFeatureValue(currentAnnot.getType().getFeatureByBaseName(splits[f])) was getFeatureValue(null) and giving an error